Detectors/FIT/FV0/base/src/Geometry.cxx

```diff
@@ -7,117 +7,1183 @@
+  mDrMinScrews.push_back(sDrMinScrewTypes[screwTypeID]);
+  mDrMaxScrews.push_back(sDrMaxScrewTypes[screwTypeID]);
+  mDzMaxScrews.push_back(sDzMaxScrewTypes[screwTypeID]);
+  mDzMinScrews.push_back(sDzMinScrewTypes[screwTypeID]);
+}
+
+void Geometry::addRodProperties(const int rodTypeID, const int iRing)
+{
+  mRodTypeIDs.push_back(rodTypeID);
+  mRodPos.push_back(std::vector<float>{sDxMinRodTypes[rodTypeID] / 2,
+                                       mRScrewAndRod[iRing],
+                                       sZScintillator - sDzScintillator / 2 + sZShiftRod + sDzMaxRodTypes[rodTypeID] / 2});
+  mDxMinRods.push_back(sDxMinRodTypes[rodTypeID]);
+  mDzMaxRods.push_back(sDxMaxRodTypes[rodTypeID]);
+  mDyMinRods.push_back(sDyMinRodTypes[rodTypeID]);
+  mDyMaxRods.push_back(sDyMaxRodTypes[rodTypeID]);
+  mDzMaxRods.push_back(sDzMaxRodTypes[rodTypeID]);
+  mDzMinRods.push_back(sDzMinRodTypes[rodTypeID]);
+
+  mRodTypeIDs.push_back(rodTypeID);
+  mRodPos.push_back(std::vector<float>{sDxMinRodTypes[rodTypeID] / 2,
+                                       -mRScrewAndRod[iRing],
+                                       sZScintillator - sDzScintillator / 2 + sZShiftRod + sDzMaxRodTypes[rodTypeID] / 2});
+  mDxMinRods.push_back(sDxMinRodTypes[rodTypeID]);
+  mDzMaxRods.push_back(sDxMaxRodTypes[rodTypeID]);
+  mDyMinRods.push_back(sDyMinRodTypes[rodTypeID]);
+  mDyMaxRods.push_back(sDyMaxRodTypes[rodTypeID]);
+  mDzMaxRods.push_back(sDzMaxRodTypes[rodTypeID]);
+  mDzMinRods.push_back(sDzMinRodTypes[rodTypeID]);
+}
+
+void Geometry::initializeScrewAndRodPositionsAndDimensions()
+{
+  for (int iRing = 0; iRing < mRScrewAndRod.size(); ++iRing) {
+    switch (iRing) {
+      case 0:
+        addRodProperties(0, iRing);
+        for (float phi = 45; phi >= -45; phi -= 45) {
+          addScrewProperties(0, iRing, phi);
+        }
+        break;
+      case 1:
+        addRodProperties(0, iRing);
+        for (float phi = 45; phi >= -45; phi -= 45) {
+          addScrewProperties(1, iRing, phi);
+        }
+        break;
+      case 2:
+        addRodProperties(1, iRing);
+        for (float phi = 67.5; phi >= -67.5; phi -= 22.5) {
+          addScrewProperties(2, iRing, phi);
+        }
+        break;
+      case 3:
+        addRodProperties(2, iRing);
+        for (float phi = 67.5; phi >= -67.5; phi -= 22.5) {
+          addScrewProperties(3, iRing, phi);
+        }
+        break;
+      case 4:
+        addRodProperties(3, iRing);
+        for (float phi = 45; phi >= -45; phi -= 45) {
+          addScrewProperties(4, iRing, phi);
+        }
+        break;
+      case 5:
+        addRodProperties(3, iRing);
+        for (float phi = 67.5; phi >= -67.5; phi -= 22.5) {
+          addScrewProperties(5, iRing, phi);
+        }
+        break;
+      default:
+        break;
+    }
+  }
+}
+
+void Geometry::initializeTransformations()
+{
+  TGeoTranslation* leftTranslation = new TGeoTranslation(-sDxHalvesSeparation / 2, 0, 0);
```

Comment:
we try to minimize new/delete, e.g. here better use local objects (self-destructed)
````
TGeoTranslation leftTranslation(-sDxHalvesSeparation / 2, 0, 0);
TGeoRotation leftRotation;
leftRotation.ReflectX(true);
TGeoHMatrix leftTotalTransformation = leftTranslation * leftRotation;
````

Reply:
Fixed as well